### DISCUSSION TITLE
Trouble with PowerShell 7.4 despite recommended solutions

### DISCUSSION THREAD

**Discussion by ggottemo** [JanDeDobbeleer/oh-my-posh#5007] [Troubleshoot] [answered]
Hello!

Even after following the FAQ and suggestions in related issues ( https://github.com/PowerShell/PowerShell/issues/20733 ,https://github.com/PowerShell/PowerShell/issues/7233#issuecomment-640243647 ), I am unable to get the prompt to properly render:
![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/16184296/31582f1c-0de4-4c3c-bfed-4a3f658a37bc)

I have tried setting the encoding as described in the FAQ at the beginning of my PowerShell Profile 
`[Console]::OutputEncoding = [Text.Encoding]::UTF8` 
I have tried the other suggestion from the issue I linked:
`[console]::InputEncoding = [console]::OutputEncoding = [System.Text.UTF8Encoding]::new()`
I have also tried the system wide fix by enabling the 'beta' utf8 encoding from the control panel - and that didn't seem to work either. The prompt only renders when I hit `CTRL+L`, and entering any other commands just displays the `PS>`. I even tried installing the powershell 5 preview hoping that maybe it would be fixed, but no luck. 

This is my profile right now - mytheme.omp.json is just one of the included ones that I customized a bit to add the Azure segment with the current subscription.
```
oh-my-posh init pwsh --config "~/.mytheme.omp.json" | Invoke-Expression

$Env:ASPNETCORE_ENVIRONMENT = "Debug"

Import-Module -Name Microsoft.WinGet.CommandNotFound
$env:POSH_AZURE_ENABLED = $true
```

Just wondering if there's something else I can try or if I'm doing something wrong here - all the related discussions I've found so far were solved by setting the encoding at the top of the profile, which didn't work for me

**Comment by JanDeDobbeleer** [comment]
Can you have a look at what the `$error` variable has as value? It might display the actual issue.

**Comment by ggottemo** [comment, marked as answer]
Thanks for that! For anyone else that ends up here, the basic answer is to double check any extra stuff you put in the config! It was filled with a bunch of errors like this: 
```
Get-Content: Cannot find drive. A drive with the name 'System.Management.Automation.CommandNotFoundException' does not exist.
Get-Content: Cannot find path 'C:\{my cwd} The term 'Get-AzContext' is not recognized as a name of a cmdlet, function, script file, or executable program.
```
After removing the CommandNotFound module line from my profile, I got it down to this:

``` 
PS>$error

ErrorRecord                 : The term 'Get-AzContext' is not recognized as a name of a cmdlet, function, script file, or
                              executable program.
                              Check the spelling of the name, or if a path was included, verify that the path is correct
                              and try again.
CommandName                 : Get-AzContext
WasThrownFromThrowStatement : False
TargetSite                  : System.Collections.ObjectModel.Collection`1[System.Management.Automation.PSObject]
                              Invoke(System.Collections.IEnumerable)
Message                     : The term 'Get-AzContext' is not recognized as a name of a cmdlet, function, script file, or
                              executable program.
                              Check the spelling of the name, or if a path was included, verify that the path is correct
                              and try again.
Data                        : {[System.Management.Automation.Interpreter.InterpretedFrameInfo,
                              System.Management.Automation.Interpreter.InterpretedFrameInfo[]]}
InnerException              :
HelpLink                    :
Source                      : System.Management.Automation
HResult                     : -2146233087
StackTrace                  :    at System.Management.Automation.Runspaces.PipelineBase.Invoke(IEnumerable input)
                                 at Microsoft.PowerShell.Executor.ExecuteCommandHelper(Pipeline tempPipeline, Exception&
                              exceptionThrown, ExecutionOptions options) 
```

Looking back at the docs, I understand my mistake. I confused the AZ **module** for powershell with the AZ **CLI tool** (the latter being the one I'm using). So, I didn't need that line in my profile at all and removing it causes the prompt to render correctly again. Not sure why there were so many errors for the command not found module, but I re-added the import line for it after removing `$env:POSH_AZURE_ENABLED` and it seems to be working fine. Thanks!
